model_analyzer/entrypoint.py

```diff
@@ -0,0 +1,210 @@
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import sys
+import time
+from itertools import product
+
+from .cli.cli import CLI
+
+from model_analyzer.model_analyzer_exceptions import TritonModelAnalyzerException
+from .monitor.nvml import NVMLMonitor
+from .perf_analyzer.perf_analyzer import PerfAnalyzer
+from .perf_analyzer.perf_config import PerfAnalyzerConfig
+from .triton.server.server_factory import TritonServerFactory
+from .triton.server.server_config import TritonServerConfig
+from .triton.client.client_factory import TritonClientFactory
+from .triton.client.client_config import TritonClientConfig
+from .triton.model.model import Model
+from .record.record_aggregator import RecordAggregator
+from .record.gpu_free_memory import GPUFreeMemory
+from .record.gpu_used_memory import GPUUsedMemory
+from .record.perf_throughput import PerfThroughput
+from .output.output_table import OutputTable
+from .output.file_writer import FileWriter
+
+
+def create_perf_config(params):
+    """
+    Utility function for creating
+    a PerfAnalyzerConfig from
+    a dict of parameters
+
+    Parameters
+    ----------
+    params : dict
+        keys are arguments to perf analyzer
+        values are their value
+
+    Returns
+    -------
+    PerfAnalyzerConfig
+        The arguments from params
+        are set in this config
+    """
+
+    config = PerfAnalyzerConfig()
+    for param, value in params.items():
+        config[param] = value
+    return config
+
+
+def main():
+    """
+    Main entrypoint of model_analyzer
+    """
+
+    # Get args
+    cli = CLI()
+    args = cli.parse(sys.argv)
+    print(args)
```

Comment:
This should be gated by a verbosity flag. We should also use logging, and not just prints